### DISCUSSION TITLE
Keep interactivity with inkscape svgs

### DISCUSSION THREAD

**Discussion by DunLug** [senshu/Sozi#601] [Q&A] [answered]
Hello everyone and thank you for your work with Sozi,

I would like to know if it is possible with sozi to use an inkscape svg that embed interactivity functionnalities and to not erase this interactivity in the final presentation.
My goal would be to show a diagram with hidden informations and to reveal it when clicking or hovering rectangles, in order to make a kind of custom quizz.
I was able to make a proof of concept with inkscape when read in Firefox, however when I tried to embed svg to Impress or Sozi, it seems that the interactivity is removed, just as if the svg was exported to a simple image.

I suppose that interactivity used to control Sozi (click to go to next frame) overrides this behaviour, while the javascript commands I have added to my svg are kept in the final html file.

Is there an option to allow this interactivity ? I know it is a specific behaviour but it would be a great functionnality for me if it is possible.

I attached an example of file with interactivity.
![test_anim](https://user-images.githubusercontent.com/6756570/152786225-6e1b873a-2684-40dc-ab4e-3fde8e1e9d85.svg)

Thanks you for your answers

**Comment by DunLug** [comment, marked as answer]
Hi everyone,
After some researches it seems that I used InkWeb library to allow interactivity. The library is embedded inside the svg file but disappear when generating the Sozi presentation. A workaround is to manually edit the HTML file to add the library definition, or to avoid using it inside interactivity fields.

**Reply by aumouvantsillage** [reply]
Hi, @DunLug 

As you have observed, Sozi removes all embedded scripts from the SVG.
The primary reason is to prevent conflicts with the JavaScript of Sozi itself, when the SVG is loaded in the editor and when the presentation is loaded in a browser.

> A workaround is to manually edit the HTML file to add the library definition

Sozi allows you to add custom scripts to the generated HTML without having to edit it manually.
I have not examined your example yet to evaluate whether this approach would work in your situation.

**Reply by DunLug** [reply]
Hi,
Thanks you very much for your answer, I'll try to check this feature as soon as possible.
Regards
